base_binary_url_import/wizard/base_binary_url_import.py

```diff
@@ -0,0 +1,343 @@
+        }
+        db_ids_set = set(db_ids)
+        xml_ids_db_ids_set = set(xml_ids_records_ids_map.values())
+        intersects = db_ids_set.intersection(xml_ids_db_ids_set)
+        if intersects:
+            raise UserError(
+                _("Following DB IDs intersect with XML IDs:\n %s")
+                % LINE_BREAK_CHAR.join([str(db_id) for db_id in intersects])
+            )
+
+    def lines_sanity_check(self):
+        """Check if IDs are existing for selected model and we don't have two
+        lines matching same record"""
+        self.ensure_one()
+        db_ids, xml_ids = self._get_db_and_xml_ids()
+        # Check DB IDs exist
+        self._check_db_ids_exist(db_ids)
+        # Check we don't have two lines targeting same DB ID
+        self._check_db_ids_duplicated(db_ids)
+        # Check XML IDs exist and match target model
+        self._check_xml_ids_exist(xml_ids)
+        # Check we don't have two lines targeting same XML ID
+        self._check_xml_ids_duplicated(xml_ids)
+        # Check no intersection between XML IDs and DB IDs
+        self._check_db_ids_xml_ids_intersecting(db_ids, xml_ids)
+
+    def action_import_lines(self):
+        self.ensure_one()
+        self.lines_sanity_check()
+        updated_ids = []
+        with requests.Session() as session:
+            for line in self.line_ids:
+                updated_ids.append(line.import_binary_from_url(session))
+        domain = [("id", "in", updated_ids)]
+        return {
+            "name": _("Updated records"),
+            "type": "ir.actions.act_window",
+            "res_model": self.target_model_id.model,
+            "domain": domain,
+            "view_mode": "tree,form",
+        }
+
+
+class BaseBinaryURLImportLine(models.TransientModel):
+
+    _name = "base.binary.url.import.line"
+    _description = "Import of binaries from URL wizard line"
+
+    wizard_id = fields.Many2one("base.binary.url.import", required=True)
+    target_record_identifier = fields.Char(string="Target record ID")
+    is_target_record_identifier_required = fields.Boolean(
+        compute="_compute_is_target_record_identifier_required"
+    )
+    binary_url_to_import = fields.Char(string="URL to import", required=True)
+
+    @api.depends("binary_url_to_import")
+    def _compute_is_target_record_identifier_required(self):
+        for line in self:
+            line.is_target_record_identifier_required = (
+                SEPARATOR_CHAR not in line.binary_url_to_import
+                if line.binary_url_to_import
+                else False
+            )
+
+    def _get_identifier_type(self):
+        self.ensure_one()
+        target_id = self.target_record_identifier
+        if not target_id:
+            return
+        if target_id.isdigit():
+            return "db_id"
+        elif "." in target_id:
+            return "xml_id"
+        else:
+            raise UserError(_("Identifier %s is not an Integer or XMLID") % target_id)
```

Comment:
```suggestion
        raise UserError(_("Identifier %s is not an Integer or XMLID") % target_id)
```

Reply:
I think we don't need to modify as your suggestion

Reply:
I approved it an anyway ;)